hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/cost/Cost.java

```diff
@@ -132,7 +132,11 @@ public Cost plus(RelOptCost other) {
 
     @Override
     public RelOptCost minus(RelOptCost other) {
-        throw new UnsupportedOperationException("Should not be called.");
+        Cost other0 = (Cost) other;
```

Comment:
Why do we need to add this when removing the distribution trait?

And if we do need it, shouldn't it be analogous to `plus`, that is, handle infinity?